crates/ink/src/codegen/dispatch/execution.rs

```diff
@@ -119,65 +113,34 @@ pub trait ConstructorReturnType<C>: private::Sealed {
     ///
     /// # Note
     ///
-    /// For infallible constructors this is `core::convert::Infallible`.
+    /// For infallible constructors this is `()`.
```

Comment:
I kinda dislike that this is now `()` for infallible errors since that means to me that we technically cannot have `()` as error type for ink! constructors. So
```rust
#[ink(constructor)]
fn foo() -> Result<Self, ()>
```
is not accepted? I would not argue that I'd like having `()` type as error type in particular but as proc. macro developers we should be generous with what input we accept. Therefore I strongly suggest having a special type (again) that indicates infallible ink! constructor "error" types defined in the `ink_lang` crate alongside other similar marker types.

Reply:
The code snippet you provided is valid and will result in successful build of a contract

Reply:
But I have returned `Infallible` back

Reply:
`Infallible` doesn't implement `Encode` so it can't be used as an error type